src/wallet/wallet_rpc_server_commands_defs.h

```diff
@@ -1906,27 +1906,44 @@ namespace wallet_rpc
     typedef epee::misc_utils::struct_init<response_t> response;
   };
 
-  struct uri_spec
+  struct uri_payment
   {
     std::string address;
-    std::string payment_id;
     uint64_t amount;
-    std::string tx_description;
     std::string recipient_name;
 
     BEGIN_KV_SERIALIZE_MAP()
-      KV_SERIALIZE(address);
-      KV_SERIALIZE(payment_id);
-      KV_SERIALIZE(amount);
-      KV_SERIALIZE(tx_description);
-      KV_SERIALIZE(recipient_name);
+    KV_SERIALIZE(address);
+    KV_SERIALIZE_OPT(amount, (uint64_t)0);
+    KV_SERIALIZE(recipient_name);
+    END_KV_SERIALIZE_MAP()
+  };
+
+  struct uri_spec
+  {
+    std::vector<uri_payment> payments;
+    std::string tx_description;
+    std::string payment_id;
+
+    BEGIN_KV_SERIALIZE_MAP()
+    KV_SERIALIZE(payments);
+    KV_SERIALIZE(tx_description);
+    KV_SERIALIZE(payment_id);
     END_KV_SERIALIZE_MAP()
   };
 
   struct COMMAND_RPC_MAKE_URI
   {
-    struct request_t: public uri_spec
+    struct request_t
     {
+      std::vector<uri_payment> payments;
+      std::string tx_description;
+      std::string payment_id;
+      BEGIN_KV_SERIALIZE_MAP()
+      KV_SERIALIZE(payments);
+      KV_SERIALIZE(tx_description);
+      KV_SERIALIZE(payment_id);
+      END_KV_SERIALIZE_MAP()
```

Comment:
```suggestion
    using request_t = uri_spec;
```

Reduce code duplication, makes it clear `request_t` is one `uri_spec`